tuf/api/metadata.py

```diff
@@ -120,11 +120,11 @@ class Metadata(Generic[T]):
     def __init__(
         self,
         signed: T,
-        signatures: Dict[str, Signature],
+        signatures: Optional[Dict[str, Signature]] = None,
         unrecognized_fields: Optional[Mapping[str, Any]] = None,
     ):
         self.signed: T = signed
-        self.signatures = signatures
+        self.signatures = signatures or {}
```

Comment:
this construct is a bit problematic in a specific edge case of the argument container being empty:

```
sigs = {}
root = Metadata(Root(), sigs)
assert sigs is root.signatures
```

so maybe I should check `is None` instead

Reply:
Yea, I think this:
`self.signatures = signatures if signatures is not None else {}`
would be better.